src/domain/service/note.ts

```diff
@@ -222,4 +226,42 @@ export default class NoteService {
 
     return await this.noteRelationsRepository.updateNoteRelationById(noteId, parentNote.id);
   };
+
+  /**
+   * Raise domain error if tools, that are in note content are not specified in tools array
+   *
+   * @param tools - editor tools that were used in a note content
+   * @param content - content of the note
+   * @todo validate tool ids
+   */
+  public async validateNoteTools(tools : Note['tools'], content: Note['content'] | Record<string, never>): Promise<void> {
+    if (isEmpty(content) && (tools.length !== 0)) {
+      return;
+    }
+
+    if (isEmpty(content)) {
+      throw (new DomainError('Note tools, used in note are not specified'));
+    }
+
+    /**
+     * Tools that are used in note
+     */
+    const noteToolNames = Array.from(content.blocks).map(block => block.type);
+
+    /**
+     * Tools that are specified in tools array
+     */
+    const specifiedNoteToolNames = Array.from(tools).map(tool => tool.name);
+
+    /**
+     * Check that all tools used in note are specified in noteToolNames array
+     */
+    const toolsAreSpicified = noteToolNames.every((toolName) => {
+      return (specifiedNoteToolNames.includes(toolName));
```

Comment:
```suggestion
      return specifiedNoteToolNames.includes(toolName);
```

Reply:
what if `specifiedNoteToolNames` has more elements than `noteToolNames`? It should throw an error in that case. Probably, `noteToolNames.every()` is not enough